File: frontend/web-admin/src/api/system/menu.ts
import { http } from "@/utils/http";
import type { ApiResult } from "@/types/api";

export interface MenuItem {
  id: number;
  parentId: number;
  path: string;
  name: string;
  component: string;
  title: string;
  icon: string;
  rank: number;
  menuType: number;
  permission: string;
  showLink: number;
  status: number;
  remark?: string;
  children?: MenuItem[];
  createTime?: string;
}

/** 树形查询 */
export const getMenuTree = (keyword?: string) => {
  return http.request<ApiResult<MenuItem[]>>("get", "/api/system/menu/tree", {
    params: keyword ? { keyword } : {}
  });
};

/** 查询单个菜单 */
export const getMenuById = (id: number) => {
  return http.request<ApiResult<MenuItem>>("get", `/api/system/menu/${id}`);
};

/** 新增菜单 */
export const createMenu = (data: Partial<MenuItem>) => {
  return http.request<ApiResult<void>>("post", "/api/system/menu", { data });
};

/** 修改菜单 */
export const updateMenu = (data: Partial<MenuItem>) => {
  return http.request<ApiResult<void>>("put", "/api/system/menu", { data });
};

/** 删除菜单 */
export const deleteMenu = (id: number) => {
  return http.request<ApiResult<void>>("delete", `/api/system/menu/${id}`);
};

File: .claude/worktrees/agent-a6e02eb5/backend/src/main/java/com/wecorp/entity/Menu.java
package com.wecorp.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("t_menu")
public class Menu {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long parentId;

    private String path;

    private String name;

    private String component;

    private String title;

    private String icon;

    @TableField("`rank`")
    private Integer rank;

    private Integer menuType;

    private String permission;

    private Integer showLink;

    private Integer status;

    private String remark;

    @TableLogic
    private Integer deleted;

    @TableField(fill = FieldFill.INSERT)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updateTime;

    @TableField(exist = false)
    private List<Menu> children;
}
